serial4j-native/src/include/BufferUtils.h

```diff
@@ -60,10 +60,10 @@ namespace BufferUtils {
      * @param buffer the buffer to free its cells.
      * @param count the number of cells to free, starting from index zero.
      */
-    static inline void freeBufferCells(void** buffer, int* count) {
-        for (int i = 0; i < *count; i++) {
-            BufferUtils::nullifyBuffer(buffer, i);
+    static inline void freeBufferCells(void** buffer, int count) {
+        for (int i = 0; i < count; i++) {
             free(buffer[i]);
+            BufferUtils::nullifyBuffer(buffer, i);
         }
```

Comment:
To understand this, refer to [this gitgist](https://gist.github.com/Scrappers-glitch/312346e1bba33fc88fa160b7302acf1d).